### DISCUSSION TITLE
Is it possible to support batching inference for fastpitch model?

### DISCUSSION THREAD

**Discussion by yuananf** [NVIDIA-NeMo/NeMo#4133] [Q&A]
I exported the Nemo fastpitch model to onnx model, I saw the inputs are ```text, pitch, pace```, seems it does not support batching inference now, is it possible for fastpitch model to support batching?

**Comment by ahkarami** [comment]
@redoctopus 
Would you please answer the above question?
Best

**Reply by redoctopus** [reply]
I'm not familiar with the ONNX exporting, @borisfom would you mind taking a look?

**Reply by ahkarami** [reply]
Thanks for your reply.
Before the model becomes an ONNX model, and independent of the TTS model type (i.e., any NeMo TTS models like Mixter TTS, Tacotron2, etc.), can we inference them via batch mode?

**Reply by imans777** [reply]
For me, I'm not necessarily looking for its ONNX export. I would be glad if we could have the batch inference in the original model, with the consideration of padding inputs, handling stop tokens, etc. I believe it can improve the model performance dramatically and better usage of GPU is guaranteed.

**Reply by redoctopus** [reply]
Ahh, my bad, I understand now. No, it's not currently supported, but I don't think there should be any complications in the way of a simple implementation.

I'll put it on the list for the near future when I (or someone else on the team) has some cycles free.

**Reply by imans777** [reply]
Hi, any updates on this? Thanks.

**Reply by redoctopus** [reply]
Not yet--I believe we will have someone free in the coming weeks to work on it, though! I'll update the thread once the ball is moving.

**Comment by imans777** [comment]
Another idea alongside batch inference is batch serving, which puts the model into a server (e.g. gRPC server) and then queues requests and tries to best match inputs at close lengths and batch them together. This can be done manually, or better, via nvidia triton, which also supports different algorithms for serving requests.
